screwdriver/store/store.go

```diff
@@ -32,20 +37,95 @@ type ResponseError struct {
 type Command struct {
 	Type string
 	Body []byte
-	Meta *api.Command
+	Spec *api.Command
 }
 
-// New returns API object
-func New(meta *api.Command) (API, error) {
+func (e ResponseError) Error() string {
+	return fmt.Sprintf("Store API %d %s", e.StatusCode, e.Reason)
+}
+
+func commandURL(spec *api.Command) (string, error) {
+	if spec.Format == "binary" {
+		url := fmt.Sprintf("%scommands/%s%%2F%s/%s", config.SDStoreURL,
+			spec.Namespace, spec.Command, spec.Version)
+		return url, nil
+	}
+	return "", fmt.Errorf("The format is not binary")
+}
+
+// New returns Store object
+func New(spec *api.Command) (Store, error) {
+	c, err := newClient(spec)
+	if err != nil {
+		return nil, err
+	}
+	return Store(c), nil
+}
+
+func newClient(spec *api.Command) (*client, error) {
+	baseURL, err := commandURL(spec)
+	if err != nil {
+		return nil, err
+	}
 	c := &client{
-		baseURL: "http://store/base/",
+		baseURL: baseURL,
 		client:  &http.Client{Timeout: timeoutSec * time.Second},
-		meta:    meta,
+		spec:    spec,
+	}
+	return c, nil
+}
+
+func handleResponse(res *http.Response) ([]byte, error) {
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, fmt.Errorf("Reading response Body from Store API: %v", err)
+	}
+
+	switch res.StatusCode / 100 {
+	case 2:
+		return body, nil
+	case 4:
+		res := new(ResponseError)
+		err = json.Unmarshal(body, res)
+		if err != nil {
+			return nil, fmt.Errorf("Unparseable error response from Store API: %v", err)
+		}
+		return nil, res
+	case 5:
+		return nil, fmt.Errorf("%v: Store API has internal server error", res.StatusCode)
+	default:
+		return nil, fmt.Errorf("Unknown error happen while communicate with Store API")
+	}
+}
+
+func reviseContentType(ct string) string {
```

Comment:
I think `parseContentType` or `normalizeContentType` is better. Also you might want to add a comment that describes what you get by this func.

Reply:
This function is difficult to understand. To add comment is nice.